### ISSUE TITLE
[Tutorial] General Usage | [教程] 一般用法

### ISSUE THREAD

**Issue by Discreater** [Towdium/JustEnoughCalculation#85]
For 1.7.10 and 1.12.2 ~ 1.16.
适用于 1.7.10 以及 1.12.2 到 1.16。

![jeca_general_usage](https://user-images.githubusercontent.com/41231596/118408894-47d13f80-b6ba-11eb-9cc3-a72c73f56e59.gif)

**Comment by Discreater** [comment]
In fact, you **don't have to** move the extruder shapes or programmed circuits from the input group to catalyst group. Because the amount of them is zero. They won't affect the calculation.
事实上**没有必要**把模头和编程电路手动放置到催化剂栏。数量是0的物品不会影响计算。（但是并不能手动设置数量为0的物品）

**Comment by Sovereignty89** [comment]
Hello @Discreater, been using this mod extensively in GTNH ever since I saw that issue suggesting to add it to the pack.
Is it possible to add the ability to add decimals? For instance, sometimes a craft needs 18L, which is 12.5% of 144L. But the moment you add a . or , it doesn't allow it. Neither for percentage nor for full amounts. Is this functionality possible?


Thank you for making it work in 1.7.10 and with GT and thank you to Towdium for the mod.

**Comment by Discreater** [comment]
I don't think we need a decimal point. There is no half or quarter items in MC, which is the same for fluid. The smallest fluid unit is `mB` (or `L` in gregtech). 

> For instance, sometimes a craft needs 18L, which is 12.5% of 144L.

If you need 18L , just type in 18. And for items, you can use percent by pressing this button.
![image](https://user-images.githubusercontent.com/41231596/123821504-088b4500-d92e-11eb-8016-7f38a5a3dc14.png)


By the way, if you see a fluid with unit `L` in the calculator GUI of the 1.7.10 version, please check this issue #88 .

**Comment by Sovereignty89** [comment]
What I do is, I bookmark for instance soldering alloy 144L, I drag it in instead of the item.
![CircuitSolderingAlloy1](https://user-images.githubusercontent.com/6663736/123863605-14750800-d92a-11eb-8ab2-8638e3bd946b.PNG)
Now this recipe only requires 72L, so I can change the amount to 50%.
Then I have another recipe that knows I get 144L from one ingot.
![CircuitSolderingAlloy2](https://user-images.githubusercontent.com/6663736/123863661-25be1480-d92a-11eb-95fa-bcab396716b5.PNG)

Now it knows that for every 2 crafts (4 circuits), it needs 1 ingot. 
The problem arises when you get amounts that are for instance 18L, which 12.5% of one ingot. So I'd like to set it up like that and have it tell me I need one ingot every 8 crafts. Makes sense? That way when I make a stack, it tells me I need 8 of something or however many I need if I could work with decimals.

How would I go about doing this like you said? Do I set it up so that 1 ingot is 144x 144L?
Like so?
![CircuitSolderingAlloy3](https://user-images.githubusercontent.com/6663736/123865122-d2e55c80-d92b-11eb-9a91-3f94996a99d2.PNG)

![CircuitSolderingAlloy4](https://user-images.githubusercontent.com/6663736/123864748-65d1c700-d92b-11eb-8ad0-800dd0a9b8b7.PNG)

![CircuitSolderingAlloy5](https://user-images.githubusercontent.com/6663736/123864910-9285de80-d92b-11eb-9cc4-641459a1d39f.PNG)

You know what, this actually does seem to work. It knows it needs 6 ingots to craft 24 (12 crafts). I think this solves my problem.

**Comment by Discreater** [comment]
[1.7.10]
Have you read this issue #88 ?
> **Real fluids in the yellow rectangle have border, and their unit is mB (or B, KB...) .**

As I said, this is **not Fluid**.
![image](https://user-images.githubusercontent.com/41231596/123958803-14374400-d9e0-11eb-9ce3-27329271f3b6.png)

You can transfer this recipe from NEI. Like the recipe of Fluid Extractor.  Or if you must customize the recipe, you need to pick the fluid by pressing the "pick label" button to search the fluid, not nei.
